packages/editor/src/extensions/outline-list/outline-list.ts

```diff
@@ -123,12 +124,41 @@ export const OutlineList = Node.create<OutlineListOptions>({
     ];
   },
   addNodeView() {
-    return ({ node, HTMLAttributes }) => {
+    return ({ editor, node, getPos, HTMLAttributes }) => {
       const ul = document.createElement("ul");
       ul.classList.add("outline-list");
       if (node.attrs.textDirection) ul.dir = node.attrs.textDirection;
       for (const key in HTMLAttributes)
         ul.setAttribute(key, HTMLAttributes[key]);
+
+      const processNestedNodes = (parentNode: Fragment, parentPos: number) => {
+        parentNode.forEach((child, offset) => {
+          const childPos = parentPos + offset + 1;
+          if (
+            child.type.name === "bulletList" ||
+            child.type.name === "orderedList"
+          ) {
+            editor.commands.command(({ tr }) => {
+              tr.setNodeMarkup(childPos, editor.schema.nodes.outlineList);
+              return true;
+            });
+          }
+          if (child.type.name === "listItem") {
+            editor.commands.command(({ tr }) => {
+              tr.setNodeMarkup(childPos, editor.schema.nodes.outlineListItem);
+              return true;
+            });
+          }
+          processNestedNodes(child.content, childPos);
+        });
+      };
+
+      if (typeof getPos === "function") {
+        processNestedNodes(node.content, getPos());
+      } else {
+        processNestedNodes(node.content, Number(getPos));
+      }
+
```

Comment:
This is the wrong place to do this. We need to convert the nodes in `toggleOutlineList` command.